main.go

```diff
@@ -33,10 +39,86 @@ func main() {
 	sync := marathon.NewMarathonSync(remote, consul)
 	go sync.Sync()
 
-	// set up routes
+	fh := &ForwardHandler{consul}
+
+	version, err := remote.Version()
+	if version >= "0.9.0" {
+		log.Info(fmt.Sprintf("detected Marathon v%s with /v2/events endpoint", version))
+		SubscribeToEventStream(config, remote, fh)
+	} else {
+		log.Info(fmt.Sprintf("detected Marathon v%s -- make sure to set up an eventSubscription for this process", version))
+		ServeWebhookReceiver(config, fh)
+	}
+}
+
+func SubscribeToEventStream(config *config.Config, m marathon.Marathon, fh *ForwardHandler) {
+	buffer := make([]byte, 1024)
+	req, err := http.NewRequest("GET", m.Url("/v2/events"), bytes.NewBuffer(buffer))
+	if err != nil {
+		log.WithError(err).Error("Could not GET /v2/events")
+		os.Exit(1)
+	}
+	req.Header.Set("Accept", "text/event-stream")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.WithError(err).Error("HTTP request for /v2/events failed!")
+		os.Exit(1)
+	}
+	defer resp.Body.Close()
+	log.Info("reader here")
+	reader := bufio.NewReader(resp.Body)
+
+	for {
+		body, err := reader.ReadBytes('\n')
+		if err != nil {
+			panic(err)
+		}
+
+		// marathon sends blank lines to keep the connection alive
+		if bytes.Equal(body, []byte{'\r', '\n'}) {
+			continue
+		}
+
+		// we don't care about these headers, since the data blob has an
+		// "eventType" field
+		if string(body[0:6]) == "event:" {
+			continue
+		}
+
+		if string(body[0:5]) == "data:" {
+			body = body[6:]
+			eventType, err := events.EventType(body)
+			if err != nil {
+				log.WithError(err).Error("error parsing event")
+				continue
+			}
+
+			switch eventType {
+			case "api_post_event", "deployment_info":
+				log.WithField("eventType", eventType).Info("handling event")
```

Comment:
I realize this pattern is elsewhere, but let's clean it up a bit: `eventLogger := log.WithField("eventType", eventType)` just before the switch and then these lines should be `eventLogger.Info("handling event")` and the opposite.